components/customs/visualizations/line-fitting/use-best-fit-curve.tsx

```diff
@@ -0,0 +1,35 @@
+import { useCallback } from 'react';
+import { polynomial } from 'regression';
+
+interface GenerateBestFitCurveInput {
+  data: Array<{
+    xAxisValue: number,
+    yAxisValue: number
+  }>
+  maximumPolynomialOrder: number;
+}
+
+interface GenerateBestFitCurveOutput {
+  xAxisValueToYAxisValue: (input: { xAxisValue: number }) => number;
+}
+
+export const useBestFitCurve = () => {
+  const generateBestFitCurve = useCallback((input: GenerateBestFitCurveInput): GenerateBestFitCurveOutput => {
+    const allXAxisValues = input.data.map(({xAxisValue}) => xAxisValue);
+    const smallestXAxisValue = Math.min(...allXAxisValues);
+
+    const bestFitEquationCoefficients = polynomial(
+      input.data.map(({ xAxisValue, yAxisValue }) => [xAxisValue - smallestXAxisValue, yAxisValue * 10_000]),
```

Comment:
Kind of frustrating but this library struggles if the minimum x value is very high and the deltas are small because of that. It also appears to struggle a bit with y axis values that are very low so I just drop these two fixes here in order to make the graph come out better.